File: 1.TwoSum/twosum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        a = dict()
        for i in range(len(nums)):
            try:
                x = a[target-nums[i]]
                return [x,i]
            except KeyError:
                a[nums[i]]=i
        return [-1,-1]

File: 1.TwoSum/twosum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        for (int i = 0; i < (int)nums.size();i++){
            if (hash.find(target-nums[i])!=hash.end()){
                return {hash[target-nums[i]],i};
            }
            else{
                hash.insert({nums[i],i});
            }
        }
        return {-1,-1};
    }
}; 
